### PR TITLE
Add test for only non-master branch repo view

### LINKED ISSUE #3181: 500 when viewing a repo without a `master` branch
- Gitea version (or commit ref): gitea/gitea:latest pulled yesterday
- Git version:git version 2.13.6 (Apple Git-96)
- Operating system: macos
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [X] Not relevant
- Log gist:

## Description
`gitea` throws a `500` if you view a repository without a master branch

To reproduce:
 - install a fresh gitea
 - create a new repo
 - `git push -u origin some-other-branch`
 - then view the repo in gitea and it throws a 500

P.S> - Loving `gitea` - thanks for everyone's hard work!

### PR DESCRIPTION
**Pull request by lunny**
As title and will fix #3181

### CODE REVIEW

**Comment by lunny** [comment]
depend on go-gitea/git#98

**Comment by lunny** [comment]
also fix #3202

**Comment by lunny** [comment]
And the error is still copy git vendor checksum error.

**Comment by lunny** [comment]
It's unnecessary to back port since https://github.com/go-gitea/gitea/issues/3204 didn't report 500.

File: models/fixtures/repo_unit.yml [hunk 1]
```diff
@@ -67,3 +67,46 @@
   type: 5
   config: "{}"
   created_unix: 946684810
+
```
Comment (ethantkoenig) [inline models/fixtures/repo_unit.yml:70]: ~nit: please remove extra line~ this has been fixed, but the *other* blank line was removed, so the comment persists

File: integrations/links_test.go [hunk 2]
```diff
@@ -150,3 +150,26 @@ func TestLinksLogin(t *testing.T) {
...
+	var links = map[string]int{
+		"/": 200,
+	}
+	for link, expectedStatus := range links {
+		req := NewRequest(t, "GET", path.Join(repoPath, link))
+		MakeRequest(t, req, expectedStatus)
+	}
+}
+
+func TestRepoLinksNoLogin(t *testing.T) {
```
Comment (ethantkoenig) [inline integrations/links_test.go]: Please add to the existing test in `integrations/repo_test.go`, instead of creating a duplicate.

**Review by ethantkoenig — CHANGES_REQUESTED** [review]

**Comment by strk** [comment]
What's up with this PR @lunny ? Can you fix Drone ?

File: integrations/repo_test.go [hunk 3]
```diff
@@ -74,3 +75,26 @@ func TestViewRepo1CloneLinkAuthorized(t *testing.T) {
 	sshURL := fmt.Sprintf("%s@%s:user2/repo1.git", setting.RunUser, setting.SSH.Domain)
 	assert.Equal(t, sshURL, link)
 }
+
+func testRepoLinksNoLogin(repoPath string, t *testing.T) {
```
Comment (ethantkoenig) [inline integrations/repo_test.go]: It would be nice to combine these tests with the existing tests above, but perhaps that can be a separate PR.

**Review by ethantkoenig — APPROVED** [review]

**Comment by lunny** [comment]
closed by #3715

**Pull request closed without merging**